lib/graphql/result_cache/key.rb

```diff
@@ -21,7 +25,11 @@ def to_s
       private
 
       def path_clause
-        @ctx.path.join('.')
+        @ctx.path.join('.') unless @ctx.nil?
+      end
+
+      def field_clause
+        @field.name unless @field.nil?
```

Comment:
This could lead to cache key collisions in the (very contrived) example of adding result caching to nested fields with the same name on different types. It's a non-issue for us at the moment since we only use result caching on top-level fields, but it's a little timebomb waiting for that contrived example to actually happen 😉 

Example: the Filter Page type has a field `topBanner`. If you were to apply result caching to only this field (and remove the enclosing result caching on the whole `filterPage` field on the query type) then `field_clause == "topBanner"`. This could lead to a collision if for example the Shoe Finder type also has a field called `topBanner` that is also result cached.

Luckily, `@field.path` also exists and returns the full path (in this case `"filterPage.topBanner"`). So I would use `path` instead of `name` in the `field_clause`.

Here is the code to reproduce my example from above:

```ruby
# in Types::FilterPage::PageType:

  top_banner_result_cache_config = On::Graphql::ResultCache::Config.call(:top_banner)
  field :top_banner, Types::FilterPage::TopBannerType, null: true do
    extension GraphQL::ResultCache::FieldExtension, top_banner_result_cache_config
  end

  def top_banner(*args)
    object.top_banner
  end

# in Types::QueryType -> remove the result cache extension for the `filter_page` field
```

Reply:
Yes! I agree, thank you for catching that case 👍

Reply:
After recent push this works.
miss:
```ruby
[28, 37] in /Users/robertkorzeniec/.rbenv/versions/2.5.8/lib/ruby/gems/2.5.0/bundler/gems/graphql-result_cache-c2ae50c5f78a/lib/graphql/result_cache/key.rb
   28:         @ctx.path.join('.') unless @ctx.nil?
   29:       end
   30:
   31:       def field_clause
   32:         byebug
=> 33:         @field.path unless @field.nil?
   34:       end
   35:
   36:       def args_clause
   37:         @args.to_h.to_a.flatten
(byebug) @field.path
"FilterPage.topBanner"
(byebug) c
GraphQL result cache key miss: GraphQL:Result::FilterPage.topBanner:en/ch/ba5b0caee7b5bc7198c8dd0108a5a52b39c68069ace2aaa36f430b79b65f0cc6
```
hit:
```ruby
[28, 37] in /Users/robertkorzeniec/.rbenv/versions/2.5.8/lib/ruby/gems/2.5.0/bundler/gems/graphql-result_cache-c2ae50c5f78a/lib/graphql/result_cache/key.rb
   28:         @ctx.path.join('.') unless @ctx.nil?
   29:       end
   30:
   31:       def field_clause
   32:         byebug
=> 33:         @field.path unless @field.nil?
   34:       end
   35:
   36:       def args_clause
   37:         @args.to_h.to_a.flatten
(byebug) @field.path
"FilterPage.topBanner"
(byebug) c
GraphQL result cache key hit: GraphQL:Result::FilterPage.topBanner:en/ch/ba5b0caee7b5bc7198c8dd0108a5a52b39c68069ace2aaa36f430b79b65f0cc6
```